src/session/mod.rs

```diff
@@ -0,0 +1,726 @@
+//! Session Initiation
+//!
+//! This module implements the `dbus-broker-session` functionality, which
+//! initiates new D-Bus sessions together with a session controller.
+
+use clap;
+use libc;
+
+use std::ffi::{OsStr, OsString};
+
+/// Command-line Interface
+///
+/// This structure provides all command-line handling, which currently means
+/// parsing command-line arguments, providing process termination, and dealing
+/// with diagnostics.
+pub struct Cli {
+    cmd: clap::Command,
+}
+
+/// Session Configuration
+///
+/// This encapsulates all parameters required to initiate a new session.
+/// Preferably, no ambient resources are needed for the session, yet for
+/// legacy compatibility to dbus-daemon this is not true if dbus-daemon is used
+/// as broker for the session.
+#[derive(Debug)]
+pub struct Config {
+    pub config_file: Option<OsString>,
+    pub dbus_broker: Option<OsString>,
+    pub dbus_daemon: Option<OsString>,
```

Comment:
Would it be clearer what this can be if we replace

```
pub dbus_broker: Option<OsString>,
pub dbus_daemon: Option<OsString>,
```

with

```
pub dbus: DBusImplementation,
```

which is defined something like:

```
pub enum DBusImplementation {
    Broker(Option<OsString>)
    Daemon(OsString)
}
```